Content.Client/_RMC14/GhostAppearance/DeadGhostVisualsSystem.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Content.Shared._RMC14.CCVar;
+using Content.Shared._RMC14.GhostAppearance;
+using Content.Shared._RMC14.Xenonids;
+using Content.Shared._RMC14.Xenonids.Damage;
+using Content.Shared.Ghost;
+using Content.Shared.Mind;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Shared.Configuration;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+using DrawDepth = Content.Shared.DrawDepth.DrawDepth;
+
+namespace Content.Client._RMC14.GhostAppearance;
+
+public sealed class DeadGhostVisualsSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _prototypes = default!;
+    [Dependency] private readonly INetConfigurationManager _netConfigManager = default!;
+
+    private readonly float _opacity = 0.5f;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<GhostComponent, PlayerAttachedEvent>(OnPlayerAttached);
+    }
+
+    private void OnPlayerAttached(Entity<GhostComponent> ent, ref PlayerAttachedEvent args)
+    {
+        UpdateGhostSprites();
+    }
+
+    private void UpdateGhostSprites()
+    {
+        var entities = EntityQueryEnumerator<RMCGhostAppearanceComponent, SpriteComponent, ActorComponent>();
+        while (entities.MoveNext(out var ghostAppearance, out var sprite, out var actor))
+        {
+            if (!_netConfigManager.GetClientCVar(actor.PlayerSession.Channel, RMCCVars.RMCGhostAppearanceFromDeadCharacter))
+                continue;
+
+            if (!TryComp<MindComponent>(ghostAppearance.MindId, out var mind))
+                continue;
+
+            var entity = mind.OwnedEntity;
+            var originalEntity = GetEntity(mind.OriginalOwnedEntity);
```

Comment:
This won't show up correctly for other players if the owned entity is not or has not ever been in PVS range of them
It would be more reliable to copy the appearance data over server-side and see if that works